lib/Sema/SemaResolveIdentifier.cpp

```
@@ -265,6 +265,115 @@ void IdentifierResolver::visit(AsmFunctionDeclStmtType &FDS) {
   FDS.getDecl()->accept(DIR);
 }
 
+void IdentifierResolver::visit(MemberExprType &M) {
+  // NameCheckByPass prevent raise a alert when M.name() is unsolved
+  // It is usually used when M.base is special case.
+  bool NameCheckByPass = false;
+  bool BaseCheckByPass = false;
+
+  auto Base = M.getBase();
+  Base->accept(*this);
+
+  {
+    Sema::SemaScope MemberAccessScope{&Actions, 0, false};
+
+    if (Base->getType() == nullptr) {
+      if (auto I = dynamic_cast<Identifier *>(Base)) {
+        auto BaseName = I->getName().str();
+        const std::vector<std::string> GlobalPassList{"block", "msg", "tx"};
+
+        if (std::find(GlobalPassList.begin(), GlobalPassList.end(), BaseName) !=
+            GlobalPassList.end()) {
+          BaseCheckByPass = true;
+        } else {
+          // TODO: we should handle `address` type as base.
```

Do we need to throw an unimplemented error message here?